src/EnergyPlus/PhaseChangeModeling/HysteresisModel.hh

```diff
@@ -0,0 +1,125 @@
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef HYSTERESISPHASECHANGE_HH_INCLUDED
+#define HYSTERESISPHASECHANGE_HH_INCLUDED
+
+#include <vector>
+
+#include <EnergyPlus.hh>
+#include <DataGlobals.hh>
+#include <OutputProcessor.hh>
+
+namespace EnergyPlus {
+
+namespace HysteresisPhaseChange {
+
+	struct PhaseChangeStates {
+		// keeping these as ints to allow output variable reporting; could refine later into enum class
+		static const int LIQUID = -2;
+		static const int MELTING = -1;
+		static const int TRANSITION = 0;
+		static const int FREEZING = 1;
+		static const int CRYSTALLIZED = 2;
+	};
+
+	extern int numHysteresisModels;
+
+	struct HysteresisPhaseChange{
+
+		std::string name;
+
+		// input parameters
+		Real64 totalLatentHeat;
+		Real64 specificHeatLiquid;
+		Real64 deltaTempMeltingHigh;
+		Real64 peakTempMelting;
+		Real64 deltaTempMeltingLow;
+		Real64 specificHeatSolid;
+		Real64 deltaTempFreezingHigh;
+		Real64 peakTempFreezing;
+		Real64 deltaTempFreezingLow;
+
+		// history and state terms
+		bool phaseChangeTransition;
+		Real64 enthOld;
+		Real64 enthNew;
+		Real64 enthRev;
+		Real64 CpOld;
+
+		// unknown
+		Real64 specHeatTransition;
+		Real64 enthalpyM;
+		Real64 enthalpyF;
+
+		virtual
+		~HysteresisPhaseChange()
+		{}
+
+		static
+		HysteresisPhaseChange *
+		factory( const std::string objectName );
+
+		Real64 getEnthalpy( Real64 T, Real64 Tc, Real64 tau1, Real64 tau2 );
+
+		Real64 getCurrentSpecificHeat( Real64, Real64, Real64, int, int & );
```

Comment:
I don't like it either.  I was doing it early on when the argument list was fluid and changing a lot, and forgot to go back and add them in.  I fixed it.